pkg/zkcertificate/certificate_simple_json.go

```diff
@@ -0,0 +1,129 @@
+package zkcertificate
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"sort"
+	"time"
+
+	"github.com/iden3/go-iden3-crypto/poseidon"
+)
+
+// SimpleJSON represents the input data for data that consists of
+// simple JSON fields: strings, numbers, and booleans.
+type SimpleJSON map[string]interface{}
+
+// FFEncode implements FFEncoder.
+func (k SimpleJSON) FFEncode() (SimpleJSONContent, error) {
+	hashedContent := make(SimpleJSONContent)
+
+	for key, value := range k {
+		var valueStr string
+
+		switch v := value.(type) {
+		case string:
+			valueStr = v
+		case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64, float32, float64:
+			valueStr = fmt.Sprintf("%v", v)
+		case bool:
+			if v {
+				valueStr = "true"
+			} else {
+				valueStr = "false"
+			}
+		case time.Time:
+			valueStr = v.Format(time.RFC3339)
+		default:
+			return nil, fmt.Errorf("unsupported type for field %s: %T", key, v)
+		}
+
+		hash, err := poseidon.HashBytes([]byte(valueStr))
+		if err != nil {
+			return nil, fmt.Errorf("hash field %s: %w", key, err)
+		}
+		hashedContent[key] = HashFromBigInt(hash)
+	}
+
+	return hashedContent, nil
+}
+
+// Validate performs validation on the SimpleJSON instance.
+// Currently, it does not perform any validation as SimpleJSON can contain any type of data.
+func (k *SimpleJSON) Validate() error {
+	// Currently, SimpleJSON does not have specific validation requirements.
+	// We may add validation rules here if needed in the future.
+	return nil
```

Comment:
Stuff that you do in `FFEncode()` suits here more. My main goal is to prevent creating an invalid value if possible